### PR TITLE
Draft contributor ladder template per work over several weeks

### PR DESCRIPTION
**Pull request by jberkus**
on Hackmd with Karen, Paris, and others.

This version does NOT have a separate maintainer roles file.  Instead, it contains a really long comment on what kinds of specialty maintainer roles you might have.  That list is taken from https://hackmd.io/UVqeZU_DQHGiNiYJZ52eJA?view .

Missing: it would be ideal to have one example of each potential maintainer role.  I've put example links there, but they're currently blank.

### CODE REVIEW

**Comment by carolynvs** [comment]
I have created a drafts branch where we can merge templates that aren't 100% finished yet. Then once they are complete, we can do a final PR to move the file to main.

**Comment by jberkus** [comment]
for wording nits, typos, etc., please "suggest changes" instead of putting them in comments.  We're going to do a squash merge at the end of this anyway.

File: contributor-ladder.md
Comment (karenhchu) [inline contributor-ladder.md:77]: tied to...?

File: contributor-ladder.md
Comment (karenhchu) [inline contributor-ladder.md:176]: Not sure where this goes but should we include something about X max number of maintainers from a project or would that go into the governance doc. asking bc i saw that you added something similar for the org maintainer process.

File: contributor-ladder.md
Comment (karenhchu) [inline contributor-ladder.md:192]:
```suggestion
* Community Manager: owns social media, community metrics, new contributor process, contributor growth and sustainability, events, and similar areas.  [Example]()
```

File: contributor-ladder.md
Comment (jberkus) [reply contributor-ladder.md:176]: yeah, probably a note

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:5]:
```suggestion
Particularly, this file provides you with a menu of options for your project.  Most projects have 3-4 defined contributor roles. We have outlined more roles below than you may need, so select what makes sense for the structure and size of your project. Other roles should be deleted, or combined into an adjacent role.
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:20]: The links are broken because they only require a single #.

```suggestion
* [Contributor Ladder](#contributor-ladder-template)
    * [Community Participant](#contributor-participant)
    * [Contributor](#contributor)
    * [Organization Member](#organization-member)
    * [Reviewer](#reviewer)
    * [Maintainer](#maintainer)
* [Inactivity](#inactivity)
* [Involuntary Removal](#involuntary-removal)
* [Stepping Down/Emeritus Process](#stepping-downemeritus-process)
* [Contact](#contact)
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:27]: We may want to change "Rights" to "Privileges".

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:35]:
```suggestion
Description: A Community Participant engages with the project and its community, contributing their time, thoughts, etc. Community participants are usually users who have stopped being anonymous and started being active in project discussions.
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:46]: The bullets are not rendering because they are over indented:

```suggestion
* How users can get involved with the community:
    * Participating in community discussions
    * Helping other users
    * Submitting bug reports
    * Commenting on issues
    * Trying out new releases
    * Attending community events
    * [TODO: Other examples of non-contribution participation]
```

File: contributor-ladder.md
Comment (carolynvs) [reply contributor-ladder.md:77]: Don't leave me hanging 😀

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:86]: Remove blank line, it alters how the list is rendered

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:110]:
```suggestion
<!-- the process of becoming an organization member is going to depend strongly on how your project manages its infrastructure. TODO: Project leads to fill in exact details of how a contributor becomes an organization member-->
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:140]: The last bullet is rendered as a code block because the comment breaks up the list.

```suggestion
    * Can recommend and review other contributors to become Reviewers

<!-- TODO: define how granting privileges works with your specific project.  For example:  "Is listed as Approver in the OWNERS file for certain directories. -->
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:155]:
```suggestion
A Maintainer must meet the responsibilities, and requirements of a Reviewer, plus:
```

File: contributor-ladder.md
Comment (carolynvs) [inline contributor-ladder.md:184]:
```suggestion
Some projects have additional, specifically defined maintainer roles because of leadership positions that don't fit within the general maintainer template above, including ones that have special requirements.  In addition to spelling out those requirements, defining and publishing additional maintainer roles can be a way to recruit maintainers in those specific areas, especially non-code maintainers.  Here are examples of special maintainer roles which have been defined by a variety of projects.  As all of these roles are highly dependent on exact project organization, tooling, etc., these roles are not templatable.
```

**Review by carolynvs — CHANGES_REQUESTED** [review]
I think some of these should be fixed before we merge because they mess up how the file renders.

File: contributor-ladder.md
Comment (leecalcote) [inline contributor-ladder.md:190]: Is there a template for the Release Manager role, yet?

Comment (jberkus) [reply contributor-ladder.md:190]: Nope, if you know of an example, then please give me a link!

File: contributor-ladder.md
Comment (xmulligan) [inline contributor-ladder.md:61]:
```suggestion
    * Regularly or semi-regularly shows up at meetings, takes notes
```

File: contributor-ladder.md
Comment (xmulligan) [inline contributor-ladder.md:66]:
```suggestion
    * Promotes the project in public through talks, Twitter, or other mediums
```

File: contributor-ladder.md
Comment (xmulligan) [inline contributor-ladder.md:79]:
```suggestion
Description: An Organization Member is an established contributor who regularly participates in the project. Organization Members have privileges in both project repositories and elections, and as such are expected to act in the interests of the whole project.
```

File: contributor-ladder.md
Comment (carolynvs) [reply contributor-ladder.md:190]: No, we don't have that yet.

Comment (jberkus) [reply contributor-ladder.md:190]: I'm just linking to the Kubernetes Release Team docs, then.

File: contributor-ladder.md
Comment (jberkus) [reply contributor-ladder.md:61]: I'm just going to remove the "regularly"

File: contributor-ladder.md
Comment (jberkus) [reply contributor-ladder.md:66]: Is there a reason why we'd name specific channels here instead of leaving it general?

File: contributor-ladder.md
Comment (jberkus) [reply contributor-ladder.md:79]: thanks

**Comment by jberkus** [comment]
OK, edited version.  I've incorporated all feedback except for the one item I queried above.

File: contributor-ladder.md
Comment (xmulligan) [reply contributor-ladder.md:65]: I thought it might be helpful to give a couple examples, but I am also fine if you leave them out

**Review by carolynvs — APPROVED** [review]
LGTM

File: contributor-ladder.md
Comment (jberkus) [reply contributor-ladder.md:65]: I'm worried that any non-exhaustive list could be looked at as a restrictive list, as in "stuff not on this list doesn't count"

**Pull request merged**
